sound/soc/intel/boards/sof_sdw_rt_dmic.c

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright (c) 2024 Intel Corporation
+
+/*
+ * sof_sdw_rt_dmic - Helpers to handle Realtek SDW DMIC from generic machine driver
+ */
+
+#include <linux/device.h>
+#include <linux/errno.h>
+#include <sound/soc.h>
+#include <sound/soc-acpi.h>
+#include "sof_board_helpers.h"
+#include "sof_sdw_common.h"
+
+static const char * const dmics[] = {
+	"rt715",
+	"rt712-sdca-dmic",
+};
+
+int rt_dmic_rtd_init(struct snd_soc_pcm_runtime *rtd)
+{
+	struct snd_soc_card *card = rtd->card;
+	struct snd_soc_component *component;
+	struct snd_soc_dai *codec_dai;
+	char *mic_name;
+
+	codec_dai = get_codec_dai_by_name(rtd, dmics, ARRAY_SIZE(dmics));
```

Comment:
I guess it is not possible to have both rt715 and rt712, or a mix if rt715 and rt715-sdca?

Reply:
> I guess it is not possible to have both rt715 and rt712, or a mix if rt715 and rt715-sdca?

No, we don't have such configuration. We have at most one SDW DMIC. But, if it really happens, `rt_dmic_rtd_init()` will be called twice, and both rt715 and rt712 will be appended to `card->components`.

Reply:
> > I guess it is not possible to have both rt715 and rt712, or a mix if rt715 and rt715-sdca?
> 
> No, we don't have such configuration. We have at most one SDW DMIC. But, if it really happens, `rt_dmic_rtd_init()` will be called twice, and both rt715 and rt712 will be appended to `card->components`.

Sorry, I was wrong. `rt_dmic_rtd_init()` will not be called twice since the `rtd_init_done` will be set.